applications/ParticleMechanicsApplication/python_scripts/particle_mechanics_analysis.py

```diff
@@ -69,22 +69,34 @@ def __init__(self, model, project_parameters):
     def RunSolutionLoop(self):
         """This function executes the solution loop of the AnalysisStage"""
         import time
+
+        ## Analysis timer start
+        analysis_start_time = time.time()
+
         while self.time < self.end_time:
-            ## Timer Start
+            ## Solution loop timer start
             start_solve_time = time.time()
 
             self.time = self._GetSolver().AdvanceInTime(self.time)
+            self._GetSolver().SearchElement()
```

Comment:
I think this could be integrated in InitializeSolutionStep

Preferably you don't have to override `RunSolutionLoop`